### PR TITLE
Firstversion

### PR DESCRIPTION
**Pull request by allankouidri**

### CODE REVIEW

File: train_huggingface_semantic_segmentation_process.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+            pretrained_model_name,
+            num_labels = self.num_labels,
+            id2label = id2label,
+            label2id = label2id,
+            ignore_mismatched_sizes=True)
+
+        # Setting up output directory
+        if param.cfg["output_folder"] is None:
+            param.cfg["output_folder"] = os.path.dirname(os.path.realpath(__file__)) +\
+                                         "/outputs/" + param.cfg["model_name"]
```
Comment (LudoBar) [inline train_huggingface_semantic_segmentation_process.py]: Use os.path.join() or os.sep instead or '/' in path

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+--find-links https://download.pytorch.org/whl/cu111/torch_stable.html
+torch == 1.9.0
+torchvision == 0.10.0
+transformers
```
Comment (LudoBar) [inline requirements.txt]: No version here?

File: train_huggingface_semantic_segmentation_process.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        semantic_seg_masks_file = []
+        for image in input.data["images"]:
+            semantic_seg_masks_file.append(image["semantic_seg_masks_file"])
+        dict_mask = {'label': semantic_seg_masks_file}
+        dataset_mask = Dataset.from_dict(dict_mask).cast_column("label", Image()) # Mask dataset
+
+        # Merging images and masks
+        dataset = datasets.concatenate_datasets([dataset_img, dataset_mask], axis=1)
+        dataset = dataset.shuffle(seed=1)
+        dataset = dataset.train_test_split(param.cfg["test_size"]) # Train/test split
```
Comment (LudoBar) [inline train_huggingface_semantic_segmentation_process.py]: What is exactly "test_size", percentage? Should we change the name in this case?

File: train_huggingface_semantic_segmentation_process.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+            param.cfg["output_folder"] = os.path.dirname(os.path.realpath(__file__)) +\
+                                         "/outputs/" + param.cfg["model_name"]
+        os.makedirs(param.cfg["output_folder"], exist_ok=True)
+
+        # Tensorboard directory
+        str_datetime = datetime.now().strftime("%d-%m-%YT%Hh%Mm%Ss")
+        tb_dir = str((Path(core.config.main_cfg["tensorboard"]["log_uri"]) / str_datetime))
+
+        # Hyperparameters and costumization settings during training
+        training_args = TrainingArguments(
```
Comment (LudoBar) [inline train_huggingface_semantic_segmentation_process.py]: Are parameters like optimizer type (SGD, ADAM...) available? If more options are available may be we can introduce a config file (yaml) to let users customize the training.

**Review by LudoBar — COMMENTED** [review]
Just read the code, I have to test now.

File: train_huggingface_semantic_segmentation_process.py [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+                                                                    )
+        self.jitter = ColorJitter(brightness=0.25, contrast=0.25, saturation=0.25, hue=0.1)
+
+        train_ds.set_transform(self.train_transforms)
+        test_ds.set_transform(self.val_transforms)
+
+        # Labels preparation
+        self.num_labels = len(input.data["metadata"]['category_names'])
+        self.id2label = input.data["metadata"]['category_names']
+        print(self.id2label)
```
Comment (ambroiseb) [inline train_huggingface_semantic_segmentation_process.py]: is this print intended?

**Pull request merged**